Set up baseline for 7.0.0-alpha

paladox opened the pull request:
* Bump SMW requirement to 7.0
* Bump MW requirement to 1.43 (match SMW 7.0)
* Bump PHP requirement to 8.1 (match SMW 7.0)
* Adds CI for MW 1.43, MW 1.44 and MW 1.45.
* Uses gesinn-it-pub/docker-compose-ci for CI
* Runs linting, phpcs and more
* Fixes some phpcs errors - although not all.

Fixes #103

paladox committed:
Modernise CI

* Adds CI for MW 1.43
* Uses gesinn-it-pub/docker-compose-ci for CI
* Runs linting, phpcs and more
* Fixes some phpcs errors - although not all.

Fixes #103

paladox committed:
Merge branch 'master' into fix-ci

paladox commented:
@gesinn-it-gea / @gesinn-it-ilm requires https://github.com/gesinn-it-pub/docker-compose-ci/pull/9 to be merged.

paladox commented:
Hmm even with that it is still throwing "Fatal error: Uncaught ExtensionDependencyError: SemanticScribunto requires Scribunto to be installed."

paladox commented:
Oh I missed a few places in https://github.com/gesinn-it-pub/docker-compose-ci/pull/9. Although I think it has to be published for things to take affect? idk.

paladox committed:
Fix

paladox commented:
Have no idea how to fix:

```
[e716f232d4cd70a1ff19f141] [no req]   MediaWiki\Extension\Scribunto\ScribuntoException: Lua error: Internal error: The interpreter has terminated with signal "5".
Backtrace:
from /var/www/html/extensions/Scribunto/includes/ScribuntoEngineBase.php(147)
#0 /var/www/html/extensions/Scribunto/includes/engines/LuaStandalone/LuaStandaloneInterpreter.php(675): MediaWiki\Extension\Scribunto\ScribuntoEngineBase->newException(string, array)
#1 /var/www/html/extensions/Scribunto/includes/engines/LuaStandalone/LuaStandaloneInterpreter.php(485): MediaWiki\Extension\Scribunto\Engines\LuaStandalone\LuaStandaloneInterpreter->handleIOError()
#2 /var/www/html/extensions/Scribunto/includes/engines/LuaStandalone/LuaStandaloneInterpreter.php(440): MediaWiki\Extension\Scribunto\Engines\LuaStandalone\LuaStandaloneInterpreter->receiveMessage()
#3 /var/www/html/extensions/Scribunto/includes/engines/LuaStandalone/LuaStandaloneInterpreter.php(326): MediaWiki\Extension\Scribunto\Engines\LuaStandalone\LuaStandaloneInterpreter->dispatch(array)
#4 /var/www/html/extensions/Scribunto/includes/engines/LuaCommon/LuaEngine.php(187): MediaWiki\Extension\Scribunto\Engines\LuaStandalone\LuaStandaloneInterpreter->registerLibrary(string, array)
#5 /var/www/html/extensions/Scribunto/includes/engines/LuaCommon/LuaEngine.php(156): Scribunto_LuaEngine->registerInterface(string, array)
#6 /var/www/html/extensions/Scribunto/includes/engines/LuaStandalone/LuaStandaloneEngine.php(23): Scribunto_LuaEngine->load()
#7 /var/www/html/extensions/Scribunto/includes/engines/LuaCommon/LuaEngine.php(235): MediaWiki\Extension\Scribunto\Engines\LuaStandalone\LuaStandaloneEngine->load()
#8 /var/www/html/extensions/Scribunto/tests/phpunit/engines/LuaCommon/LuaEngineTestHelper.php(69): Scribunto_LuaEngine->getInterpreter()
#9 /var/www/html/extensions/Scribunto/tests/phpunit/engines/LuaCommon/LuaEngineTestBase.php(72): Scribunto_LuaEngineTestBase::makeSuite(string)
#10 [internal function]: Scribunto_LuaEngineTestBase::suite(string)
#11 /var/www/html/vendor/phpunit/phpunit/src/Framework/TestSuite.php(462): ReflectionMethod->invoke(NULL, string)
#12 /var/www/html/vendor/phpunit/phpunit/src/Framework/TestSuite.php(481): PHPUnit\Framework\TestSuite->addTestFile(string)
#13 /var/www/html/vendor/phpunit/phpunit/src/Util/Configuration.php(1034): PHPUnit\Framework\TestSuite->addTestFiles(array)
#14 /var/www/html/vendor/phpunit/phpunit/src/Util/Configuration.php(910): PHPUnit\Util\Configuration->getTestSuite(DOMElement, array)
#15 /var/www/html/vendor/phpunit/phpunit/src/TextUI/Command.php(973): PHPUnit\Util\Configuration->getTestSuiteConfiguration(string)
#16 /var/www/html/vendor/phpunit/phpunit/src/TextUI/Command.php(203): PHPUnit\TextUI\Command->handleArguments(array)
#17 /var/www/html/tests/phpunit/phpunit.php(101): PHPUnit\TextUI\Command->run(array, boolean)
#18 /var/www/html/tests/phpunit/phpunit.php(153): PHPUnitMaintClass->execute()
#19 {main}
```

paladox commented:
@gesinn-it-ilm @gesinn-it-gea not sure how to resolve ^? Unless it doesn't work on my Mac in docker. (I have a M1 Mac).

The error in GitHub ci is because the https://github.com/gesinn-it-pub/docker-compose-ci/pull/9 needs merging. For some reason it doesn't use the current build checkout.

paladox commented:
```
MediaWiki has been successfully installed. You can now visit <http://localhost:8080> to view your wiki. If you have questions, check out our frequently asked questions list: <https://www.mediawiki.org/wiki/Special:MyLanguage/Manual:FAQ> or use one of the support forums linked on that page.
Fatal error: Uncaught ExtensionDependencyError: SemanticScribunto requires Scribunto to be installed.
 in /var/www/html/includes/registration/ExtensionRegistry.php:432
Stack trace:
#0 /var/www/html/includes/registration/ExtensionRegistry.php(276): ExtensionRegistry->readFromQueue(Array)
#1 /var/www/html/includes/Setup.php(278): ExtensionRegistry->loadFromQueue()
#2 /var/www/html/maintenance/doMaintenance.php(83): require_once('/var/www/html/i...')
#3 /var/www/html/maintenance/update.php(319): require_once('/var/www/html/m...')
#4 {main}
  thrown in /var/www/html/includes/registration/ExtensionRegistry.php on line 432
make: *** [build/Makefile:100: .install] Error 255
```
is still happening. I dunno if there's suppose to be a deployment of that change (CI?)

paladox committed:
Update build

gesinn-it-gea commented:
For another repo I implemented a workflow to schedule automatic updates of docker-compose-ci: https://github.com/gesinn-it-pub/IDProvider/blob/master/.github/workflows/update-submodules.yml

If you like this approach, you can add it to other repos as well. Instead of running this only on one day, we can run it daily

gesinn-it-gea commented:
I'm not using SemanticScribuntu. Maybe the way it is installed by docker-compose-ci needs to be changed (e.g. using composer to get the deps). Or make the deps being installable via docker-compose-ci

paladox commented:
> I'm not using SemanticScribuntu. Maybe the way it is installed by docker-compose-ci needs to be changed (e.g. using composer to get the deps). Or make the deps being installable via docker-compose-ci

I’m not sure. It works locally hence the error about lua returning that status number.

paladox commented:
I see that REL1_39 isn’t being supplied to where it fetches Scribunto which would be a problem.

paladox committed:
Update ci.yml

paladox committed:
Update ci.yml

paladox committed:
Update ci.yml

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Fix phpcs

paladox committed:
Update phpunit.xml.dist

paladox committed:
Update extension.json

paladox commented:
For some reason phpunit is now in an infinite loop. I don't know why.

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update extension.json

paladox committed:
Update I18nJsonFileIntegrityTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox commented:
@gesinn-it-ilm dunno how we fix:

```
1) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "ask-001.json" ('/var/www/html/extensions/Sema...1.json')
RuntimeException: Can't create user on real database
/var/www/html/tests/phpunit/includes/TestUser.php:35
/var/www/html/tests/phpunit/includes/TestUser.php:42
/var/www/html/tests/phpunit/includes/TestUserRegistry.php:82
/var/www/html/tests/phpunit/MediaWikiIntegrationTestCase.php:276
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/SMWIntegrationTestCase.php:84
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:67
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:63
/var/www/html/extensions/SemanticScribunto/tests/phpunit/Integration/JSONScript/SemanticScribuntoJsonTestCaseScriptRunnerTest.php:33
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/SMWIntegrationTestCase.php:203
```

Seems related to your change switching to MW integration class. 


Also dunno how we fix:

```
7) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "info-001.json" ('/var/www/html/extensions/Sema...1.json')
RuntimeException: Tracking is already enabled
/var/www/html/includes/libs/rdbms/ChangedTablesTracker.php:29
/var/www/html/tests/phpunit/MediaWikiIntegrationTestCase.php:694
/var/www/html/tests/phpunit/MediaWikiIntegrationTestCase.php:624
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/SMWIntegrationTestCase.php:203
```

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox commented:
@gesinn-it-gea / @gesinn-it-ilm looks like https://github.com/wikimedia/mediawiki/blob/17e259ef2475f56635cf3c4651c81606f8be62bb/tests/phpunit/includes/TestUser.php#L35 

So wgDBprefix isn't unittest_. I don't know how to fix that. Looks like SMW needs its test fixing which is out of my experience (I would have a clue how to do it). There's https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/tests/phpunit/Utils/Connection/TestDatabaseTableBuilder.php#L71 and https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/tests/phpunit/Utils/Connection/TestDatabaseTableBuilder.php#L237

paladox commented:
https://github.com/SemanticMediaWiki/SemanticMediaWiki/commit/3f6d15433efc2de8f3c365cafc7be9621f3ce9e9#diff-04a0eaaa6a686fdb4e98911c7ce0f7f294f06950dcb3839cabaac007ba86f3ccL29

Looks like we re-added it under the new name cc @gesinn-it-gea / @gesinn-it-ilm . I guess things need to be fixed to conform to MW Integration test class?

gesinn-it-gea commented:
Maybe the whole JSONScript runner need to be updated. Compare with other places where we use it, e.g. https://github.com/SemanticMediaWiki/SemanticDrilldown/blob/master/tests/phpunit/Integration/JSONScript/JsonTestCaseScriptRunnerTest.php or https://github.com/SemanticMediaWiki/SemanticResultFormats/tree/master/tests/phpunit/Integration/JSONScript

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox commented:
Doesn't fix the problem. MW 1.39 it's just a infinite recursion and in mw 1.42 it at least show the errors (the same one).

paladox committed:
Merge branch 'master' into fix-ci

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update extension.json

paladox committed:
Update composer.json

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update composer.json

paladox committed:
Update bootstrap.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Merge branch 'master' into fix-ci

paladox committed:
Update ci.yml

paladox committed:
Merge branch 'master' into fix-ci

paladox committed:
Update ci.yml

paladox committed:
Update ci.yml

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update ScribuntoLuaLibraryTest.php

paladox committed:
Update LibraryFactoryTest.php

paladox committed:
Update LibraryFactoryTest.php

paladox committed:
Update extension.json

paladox committed:
Update composer.json

paladox committed:
Update extension.json

paladox committed:
Update extension.json

paladox commented:
I get:

```
1) SMW\Scribunto\Tests\Unit\LuaAskResultProcessorTest::testGetProcessedResult
Failed asserting that Array &0 (
    0 => Array &1 (
        '' => null
    )
) is identical to 1.

/var/www/html/extensions/SemanticScribunto/tests/phpunit/Unit/LuaAskResultProcessorTest.php:81

2) SMW\Scribunto\Tests\Unit\LuaAskResultProcessorTest::testGetDataFromQueryResultRow
Failed asserting that Array &0 (
    '' => null
) is identical to 1.

/var/www/html/extensions/SemanticScribunto/tests/phpunit/Unit/LuaAskResultProcessorTest.php:105

3) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "ask-001.json" ('/var/www/html/extensions/Sema...1.json')
Failed "#0 verify output for the `[[Category:ask-001]] [[Has text::+]]` query"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:smw|ask| +|?#-=page|?Has text |limit=3 |mainlabel=-}}
</p></div>
==== (StringContains expected) ====
[ <li> page: Scribunto/ask/001/0</li> ], [ <li> Has text: Is a Scribunto example text</li> ]

Failed asserting that 0 matches expected 2.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

4) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "ask-002.json" ('/var/www/html/extensions/Sema...2.json')
Failed "#0 verify ask result order w/o sort and order parameter"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:smw|ask| + +<span class="smw-highlighter" data-type="4" data-state="inline" data-title="Warning" title="&quot;+&quot; is not a number."><span class="smwtticon warning"></span><span class="smwttcontent">"+" is not a number.</span></span>|?#-=page|mainlabel=-}}
</p></div>
==== (StringContains expected) ====
[ <li> This is result #1
<ul><li> page: Scribunto/ask/002/1</li></ul></li> ], [ <li> This is result #2
<ul><li> page: Scribunto/ask/002/2</li></ul></li> ], [ <li> This is result #3
<ul><li> page: Scribunto/ask/002/3</li></ul></li> ], [ <li> This is result #4
<ul><li> page: Scribunto/ask/002/4</li></ul></li> ], [ <li> This is result #5
<ul><li> page: Scribunto/ask/002/5</li></ul></li> ]

Failed asserting that 0 matches expected 5.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

5) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "ask-003.json" ('/var/www/html/extensions/Sema...3.json')
Failed "#0 mw.smw.ask with format=debug"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:smw|ask| +|?#-=page|?Has text |limit=3 |mainlabel=- |format=debug }}
</p></div>
==== (StringContains expected) ====
[ &#91;&#91;Category:Ask-003]] &#91;&#91;Has text::+]] ]

Failed asserting that 0 matches expected 1.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

6) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "ask-004.json" ('/var/www/html/extensions/Sema...4.json')
Failed "#0 verify output for the `[[Has text::+]]|?Category:ask-004=member` query"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:smw|ask|test|?#-=page|?has text=text|?Category:Ask-004|?Category:INVALID|limit=3|mainlabel=-}}
</p></div>
==== (StringContains expected) ====
[ <li> page: Scribunto/ask/004/0</li> ], [ <li> text: test</li> ], [ <li> Ask-004: true</li> ], [ <li> INVALID: false</li> ]

Failed asserting that 0 matches expected 4.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

7) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "getQueryResult-001.json" ('/var/www/html/extensions/Sema...1.json')
Failed "#0 verify output for the `[[Category:getQueryResult-001]] [[Has text::+]]` query"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:SMWgetQueryResult|getQueryResultPrintRequests| +|?Has text |limit=3 }}
</p></div>
==== (StringContains expected) ====
[ <li> fulltext: Scribunto/getQueryResult/001/1</li> ], [ <li> Has text: Is a Scribunto example text</li> ]

Failed asserting that 0 matches expected 2.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

8) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "getQueryResult-002.json" ('/var/www/html/extensions/Sema...2.json')
Failed "#0 verify getQueryResult result order w/o sort and order parameter"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:SMWgetQueryResult|getQueryResultMainLabels| + +<span class="smw-highlighter" data-type="4" data-state="inline" data-title="Warning" title="&quot;+&quot; is not a number."><span class="smwtticon warning"></span><span class="smwttcontent">"+" is not a number.</span></span>|?#-=page|mainlabel=-}}
</p></div>
==== (StringContains expected) ====
[ <li> This is result #1
<ul><li> fulltext: Scribunto/getQueryResult/002/1</li></ul></li> ], [ <li> This is result #2
<ul><li> fulltext: Scribunto/getQueryResult/002/2</li></ul></li> ], [ <li> This is result #3
<ul><li> fulltext: Scribunto/getQueryResult/002/3</li></ul></li> ], [ <li> This is result #4
<ul><li> fulltext: Scribunto/getQueryResult/002/4</li></ul></li> ], [ <li> This is result #5
<ul><li> fulltext: Scribunto/getQueryResult/002/5</li></ul></li> ]

Failed asserting that 0 matches expected 5.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

9) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "info-001.json" ('/var/www/html/extensions/Sema...1.json')
Failed "#0: a warning"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:smw|info|an error text|warning}}
</p></div>
==== (StringContains expected) ====
[ <span class="smw-highlighter" data-type="4" data-state="inline" data-title="Warning" title="an error text"> ], [ <span class="smwtticon warning"> ], [ <span class="smwttcontent">an error text</span> ]

Failed asserting that 0 matches expected 3.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

10) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "set-001.json" ('/var/www/html/extensions/Sema...1.json')
#0 storing of text property `has text` Counted properties include: [
    "_INST",
    "_MDAT",
    "_SKEY"
]
Failed asserting that actual size 3 matches expected size 4.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:212
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:118
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:79
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

11) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "set-002.json" ('/var/www/html/extensions/Sema...2.json')
#0 with error 'invalid number type' Counted properties include: [
    "_MDAT",
    "_SKEY"
]
Failed asserting that actual size 2 matches expected size 4.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:212
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:118
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:79
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

12) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "set-003.json" ('/var/www/html/extensions/Sema...3.json')
#0 Counted properties include: [
    "_MDAT",
    "_SKEY"
]
Failed asserting that actual size 2 matches expected size 3.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:212
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:118
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:79
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

13) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "set-004.json" ('/var/www/html/extensions/Sema...4.json')
#0 Counted properties include: [
    "_MDAT",
    "_SKEY"
]
Failed asserting that actual size 2 matches expected size 3.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:212
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:118
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:79
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

14) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "set-005.json" ('/var/www/html/extensions/Sema...5.json')
#0 false Counted properties include: [
    "_MDAT",
    "_SKEY"
]
Failed asserting that actual size 2 matches expected size 3.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:212
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:118
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:79
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

15) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "set-006.json" ('/var/www/html/extensions/Sema...6.json')
#1 semantic data via mw.smw.set Counted properties include: [
    "_MDAT",
    "_SKEY"
]
Failed asserting that actual size 2 matches expected size 6.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:212
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:118
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:79
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

16) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "set-007.json" ('/var/www/html/extensions/Sema...7.json')
#0 semantic data via mw.smw.set Counted properties include: [
    "_MDAT",
    "_SKEY"
]
Failed asserting that actual size 2 matches expected size 4.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:212
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:118
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:79
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

17) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "set-008.json" ('/var/www/html/extensions/Sema...8.json')
Failed "#0 semantic data foobar to mw.smw.set"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:Issue75|set}}
</p></div>
==== (StringContains expected) ====
[ class="mw-parser-output" ], [ class="scribunto-error" ], [ Lua error in mw.smw.lua at line 85: Invalid parameter type supplied to smw.set(). ]

Failed asserting that 0 matches expected 3.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

18) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "subobject-001.json" ('/var/www/html/extensions/Sema...1.json')
Failed "#0, sub-test #0 storing of one subobject (text output)"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:smw|subobject|Has text=Is a Scribunto example subobject test|Has number=1}}
</p></div>
==== (StringContains expected) ====
[ Your data was stored successfully in a subobject ]

Failed asserting that 0 matches expected 1.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

19) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "subobject-002.json" ('/var/www/html/extensions/Sema...2.json')
#0 semantic data via mw.smw.subobject Counted properties include: []
Failed asserting that actual size 0 matches expected size 3.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/SemanticDataValidator.php:212
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:118
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:79
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

20) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "subobject-003.json" ('/var/www/html/extensions/Sema...3.json')
Failed "#0 semantic data foobar to mw.smw.subobject"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:Issue75|subobject}}
</p></div>
==== (StringContains expected) ====
[ class="mw-parser-output" ], [ class="scribunto-error" ], [ Lua error in mw.smw.lua at line 93: Invalid parameter type supplied to smw.subobject(). ]

Failed asserting that 0 matches expected 3.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:324
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:193

--

There were 7 skipped tests:

1) SMW\Scribunto\Tests\Unit\ScribuntoLuaLibraryAskTest
interpreter for LuaSandbox is not available

/var/www/html/extensions/Scribunto/tests/phpunit/Engines/LuaCommon/LuaEngineTestSkip.php:25

2) SMW\Scribunto\Tests\Unit\ScribuntoLuaLibraryInfoTest
interpreter for LuaSandbox is not available

/var/www/html/extensions/Scribunto/tests/phpunit/Engines/LuaCommon/LuaEngineTestSkip.php:25

3) SMW\Scribunto\Tests\Unit\ScribuntoLuaLibraryPropertyTest
interpreter for LuaSandbox is not available

/var/www/html/extensions/Scribunto/tests/phpunit/Engines/LuaCommon/LuaEngineTestSkip.php:25

4) SMW\Scribunto\Tests\Unit\ScribuntoLuaLibraryResultsTest
interpreter for LuaSandbox is not available

/var/www/html/extensions/Scribunto/tests/phpunit/Engines/LuaCommon/LuaEngineTestSkip.php:25

5) SMW\Scribunto\Tests\Unit\ScribuntoLuaLibrarySetTest
interpreter for LuaSandbox is not available

/var/www/html/extensions/Scribunto/tests/phpunit/Engines/LuaCommon/LuaEngineTestSkip.php:25

6) SMW\Scribunto\Tests\Unit\ScribuntoLuaLibrarySubobjectTest
interpreter for LuaSandbox is not available

/var/www/html/extensions/Scribunto/tests/phpunit/Engines/LuaCommon/LuaEngineTestSkip.php:25

7) SMW\Scribunto\Tests\Unit\ScribuntoLuaLibraryTest
interpreter for LuaSandbox is not available

/var/www/html/extensions/Scribunto/tests/phpunit/Engines/LuaCommon/LuaEngineTestSkip.php:25
```

paladox committed:
Update LuaAskResultProcessorTest.php

paladox committed:
Update LuaAskResultProcessorTest.php

paladox committed:
Update LuaAskResultProcessorTest.php

paladox committed:
Update LuaAskResultProcessor.php

paladox committed:
Update LuaAskResultProcessor.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update Makefile

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update ask-001.json

paladox committed:
Update ask-001.json

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Merge remote-tracking branch 'origin/master' into fix-ci

paladox committed:
Update ci.yml

paladox committed:
Update ScribuntoLuaLibrary.php

paladox commented:
@coderabbitai can you find why:

```
1) SMW\Scribunto\Integration\JSONScript\SemanticScribuntoJsonTestCaseScriptRunnerTest::testCaseFile with data set "ask-001.json" ('/var/www/html/extensions/Sema...1.json')
Failed "#0 verify output for the `[[Category:ask-001]] [[Has text::+]]` query"

==== (actual) ====
<div class="mw-content-ltr mw-parser-output" lang="en" dir="ltr"><p>{{#invoke:smw|ask| +|?#-=page|?Has text |limit=3 |mainlabel=-}}
</p></div>
==== (StringContains expected) ====
[ <li> page: Scribunto/ask/001/0</li> ], [ <li> Has text: Is a Scribunto example text</li> ]

Failed asserting that 0 matches expected 2.

/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:76
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/Validators/StringValidator.php:31
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:186
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/Utils/JSONScript/ParserTestCaseProcessor.php:83
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:323
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptServicesTestCaseRunner.php:113
/var/www/html/extensions/SemanticMediaWiki/tests/phpunit/JSONScriptTestCaseRunner.php:183
```

breaks? And can you tell me how to fix it? Seems {{#invoke:smw|ask| +|?#-=page|?Has text |limit=3 |mainlabel=-}} isn't being executed and thus returning as is.

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox committed:
Update SemanticScribuntoJsonTestCaseScriptRunnerTest.php

paladox commented:
@oetterer I'm wondering if you know how to fix the errors please? I see you did a CI change in https://github.com/SemanticMediaWiki/SemanticScribunto/commit/e06b9033cc875668eb6353ecaac87ff8177a1b85 but I'm not sure how to fix the errors.

alistair3149 committed:
Restore Scribunto #invoke parser function in JSON-script tests

The Scribunto JSON-script integration tests started rendering `{{#invoke:...}}`
literally instead of executing it, causing 18 failures across set-*, subobject-*
and ask-* fixtures. Diagnosed empirically: inside the test environment
the parser has only 87 function hooks registered (SMW's `ask`, `set` etc.
present, but `invoke` absent) and `ParserFirstCallInit` has only SMW's handler.

Root cause is in SMW's test infrastructure. `JSONScriptTestCaseRunner::setUp()`
calls `MwHooksHandler::deregisterListedHooks()`, which issues
`HookContainer::clear('ParserFirstCallInit')` to clear SMW's handler. But
`clear()` wipes *all* handlers for that hook — including Scribunto's. The
subsequent `invokeHooksFromRegistry()` only restores SMW's, so Scribunto's
`#invoke` registration never comes back.

Workaround scoped to this test class: re-register Scribunto's
`ParserFirstCallInit` handler in `setUp()` after `parent::setUp()` runs.
A proper fix belongs upstream in SMW's `MwHooksHandler`.

Also updates set-008.json and subobject-003.json to drop the now-stale
`class="mw-parser-output"` / `class="scribunto-error"` exact-attribute
matches — MW 1.44 emits combined class lists (`mw-content-ltr mw-parser-output`,
`scribunto-error mw-scribunto-error-<hash>`), so the assertions now look for
the bare class tokens.

Verified locally against MW 1.44 / PHP 8.3 / SMW dev-master: 17 of 18 JSON
fixtures pass individually (was 0 of 18). The remaining failure, ask-003,
is an unrelated pre-existing TypeError in `LibraryFactory::newQueryResultFrom`.

alistair3149 committed:
Fix TypeError on format=debug queries from mw.smw.ask

`LibraryFactory::newQueryResultFrom` was declared as `: QueryResult`, but
SMW's `Store::getQueryResult` returns a debug-output string instead of a
QueryResult for `format=debug` queries (and a few other introspection
formats). With strict typing this triggered a TypeError on the first
debug-format Lua call, aborting the test without running `tearDown` and
cascading "Tracking is already enabled" errors through every subsequent
test in the JSON-script suite.

Two changes:

* Relax `newQueryResultFrom`'s return type to `QueryResult|string` and
  document the string return path. Reverts part of 3b96db2's
  over-tightening — the existing `ScribuntoLuaLibrary::isAQueryResult`
  guard already accommodates the string path, so callers don't need
  further changes.
* In `ScribuntoLuaLibrary::ask`, move the `isAQueryResult` guard above
  the `getQuery()` member access so a string return doesn't blow up
  before it can be returned to Lua. The `getQueryResult` Lua function
  already had the guard in the right place.

Verified locally against MW 1.44 / PHP 8.3 / SMW dev-master: the full
JSON-script suite now reaches 222 assertions (was 20). The remaining
errors are pre-existing test-isolation cascades unrelated to this fix.

alistair3149 committed:
Make Scribunto PFCI re-registration version-agnostic

The previous fix in 20ea24b hardcoded a 4-arg constructor call to
MediaWiki\Extension\Scribunto\Hooks. That works on MW 1.43/1.44 but fails
on MW 1.45 where the constructor signature gained a 5th service —
ArgumentCountError cascades through every JSON-script test that follows.

Read the handler spec from Scribunto's own extension.json at setUp time
and let ObjectFactory inject whatever services the current version
declares. No version conditionals, no constructor coupling — if Scribunto
adds or drops a service in a future MW, this test follows along.

alistair3149 committed:
Align major version with Semantic MediaWiki

Bump version to 7.0.0-alpha so this extension's major matches the major
of the SMW release it targets. Versions 3.x through 6.x are intentionally
skipped — they were never released, and the jump is a one-time
realignment. Future majors follow SMW's.

Why: SemanticScribunto is a thin bridge into SMW's API surface, and every
SMW major has historically broken at least one hook or service we depend
on. Aligning the major makes wiki-version compatibility self-evident
without consulting composer constraints, and lets us freeze old majors
on the matching SMW major rather than carrying multi-SMW-version
compatibility shims in master.

Updates:
* extension.json version: 2.4.0-alpha → 7.0.0-alpha
* composer.json branch-alias: 2.4.0-dev → 7.0.0-dev
* RELEASE-NOTES.md: add 7.0.0 section explaining the realignment and
  summarising this PR's content

alistair3149 committed:
Require Semantic MediaWiki >= 7.0

Tighten the SMW dependency to match this extension's major version. With
the 7.x-tracks-SMW-7 convention from the previous commit, supporting SMW
6.x from a 7.x release of this extension would defeat the purpose — old
SMW majors should pin to the matching SemanticScribunto major.

alistair3149 committed:
Drop Travis and Scrutinizer

Neither has been doing useful work for years.

* .travis.yml: matrix tops out at MediaWiki 1.36 / PHP 7.4 and references
  shell scripts in tests/travis/ that haven't tracked the project's real
  requirements in a long time. Travis CI itself has been paid-only for
  OSS since 2021. The GitHub Actions workflow has fully covered what
  Travis used to do.
* .scrutinizer.yml: has been ERRORing on master and on every PR. The
  static-analysis it covered (phpcs, phpmd, code-style checks) is now
  run inside the docker-compose-ci flow via composer analyze.

Removes the corresponding badges and references from README.md.

alistair3149 committed:
Add Codecov badge

The GitHub Actions workflow uploads coverage to Codecov on every push;
this surfaces the current coverage figure in the README header next to
the build status badge.

alistair3149 committed:
Update README requirements and install example for 7.x

Match the new minimum versions enforced by extension.json and composer.json
(PHP 8.1, MediaWiki 1.43, SMW 7.0) and bump the composer.local.json
install example to ~7.0.

alistair3149 committed:
Tighten 7.0.0 release notes to user-facing changes only

Drop CI matrix details and the JSON-script integration-test fix from the
changelog — neither affects users running the extension. The LibraryBase
swap is an internal refactor with no behavioural change, so also dropped.

What remains is the version-alignment note, the three minimum-version
bumps that affect installability, and the mw.smw.ask format=debug
TypeError fix (which has been a fatal error on master for any user
calling that combination since early 2025).
